### ISSUE TITLE
Cannot save database to google drive on Fedora Workstation 38

### ISSUE THREAD

**Issue by vajranaga** [keepassxreboot/keepassxc#9434]
## Overview
Cannot save database to google drive on Fedora Workstation 38

## Steps to Reproduce
[NOTE]: # ( Provide a simple set of steps to reproduce this bug. )
1. add google account via gnome-settings
2. (use Files to enter into gdrive directory to 'init' otherwise keepassxc cannot read it)
3. open database
4. make some changes
5. attempt to save
6. KPXC returns error.

I have tried with multiple options:
-checked 'Use alternative saving method...', tried 'Temporary file moved into place' and 'Directly write to database file'
-unchecked 'Use alternative saving method...'

Happens with auto-saves and manual saves.

## Expected Behavior
database saves

## Actual Behavior
I get a warning or error stating the database could not be saved, and the gdrive is unmounted after keepassxc attempts to save.

## Debug Info
KeePassXC - Version 2.7.4
Revision: 63b2394

Qt 5.15.9
Debugging mode is disabled.

Operating system: Fedora Linux 38 (Workstation Edition)
CPU architecture: x86_64
Kernel: linux 6.2.14-300.fc38.x86_64

Enabled extensions:
- Auto-Type
- Browser Integration
- SSH Agent
- KeeShare
- YubiKey
- Secret Service Integration

Cryptographic libraries:
- Botan 2.19.3

## OS info
Operating System: Linux
Desktop Env: Gnome 44.1
Windowing System: Wayland

**Comment by droidmonkey** [comment]
You have to use the direct write saving option. It should work as it has been tested. If not then there might be other issues. Are you using the flatpak or distro release?

**Comment by vajranaga** [comment]
As stated in the original post, I have tried direct saving, and every other option available within keepassxc, nothing will work. And every time it unmounts the gdrive, which shouldn't happen at all under any option.

I am using the distro release installed via dnf.

**Comment by droidmonkey** [comment]
Since the gdrive is unmounted, the error seems to be happening outside of keepassxc

**Comment by vajranaga** [comment]
I have tried every available option in keepassxc, I have removed and reinstalled, it was the first app installed after reformatting and installing Fedora 38 and has been unable to save to google drive once in the few weeks I've been dealing with it. I can work with files from the google drive appropriately otherwise, and at the moment have to save my database locally and manually copy it to and from my google drive. I really don't think that is intended behaviour.

Please let me know what else I can do or provide to assist in getting this fixed.

**Comment by vajranaga** [comment]
This issue still exists. Also, it is not a KDE issue, since I am not using KDE, but Gnome.

**Comment by 9Marin** [comment]
I have the same issue. It worked fine since ~January, and then suddenly 3 weeks ago it stopped working. I also use Fedora with Gnome.

**Comment by CrezTheHomie** [comment]
Hello. I had the same confusion and found a solution that may help anyone who comes across this comment in the future.

**Steps I took**

add google account via gnome-settings
dnf install keepassxc
use Nautilus (Files) to enter into gdrive directory
`Control+C` to copy 
`Control+M` to make a symbolic link outside of google drive mount (~/Documents/.secrets/Passwords.kdbx for me)
Open database via the symbolic link.
Make a change and save.
Saves as expected.

------------------------------------

Nautilus: 45.2.1
NAME="Fedora Linux"
VERSION="39 (Workstation Edition)"
$XDG_SESSION_TYPE=wayland

-----------------------------------

The symbolic link helps with the gvfs translation from hashed file names to their "pretty names". In X11, I used to rely on GDK_ACTION_LINK for drag and drop context menu's, but that has not been implemented in Wayland.

It does not seem easy to save directly to the google drive mount, but the sym link helps.

**Comment by kevinrineer** [comment]
I am not having this problem with "Directly write to database" checked and working on a symlink local to my filesystem (made with Ctrl+C and Ctrl+M). 

Gnome: 45.2.1
Fedora Linux 39 (Workstation Edition)
keepassxc.x86_64                      2.7.6-3.fc39  installed via dnf
Windowing System: Wayland

**Comment by droidmonkey** [comment]
That is the answer, unfortunately the way gnome implements Google drive is only compatible with  direct write saving.

**Comment by heebinho** [comment]
Tried the symlink approach described by @kevinrineer,

- with the alternative saving method "Directly write to database file (dangerous)"
- without the alternative saving method

I was able to read the database and see my passwords. But saving failed with both approaches.
⚠ When saving without the alternative saving method the database was renamed and **permanently deleted** on Drive!

So be careful with this approach. Adding a comment to a closed issue only to inform.
